flake.nix

```diff
@@ -188,9 +184,21 @@
         (mapAttrs' (name: check: nameValuePair "packages/${name}" check) nonBrokenPkgs)
         // {
           inherit (self.packages.${system}) overview;
-          formatting = treefmtEval.config.build.check self;
+          pre-commit-check = pre-commit-hooks.lib.${system}.run {
+            src = ./.;
+            hooks = {
+              alejandra.enable = true;
+            };
+          };
         };
 
+      devShell.${system} = pkgs.mkShell {
+        inherit (self.checks.${system}.pre-commit-check) shellHook;
+        buildInputs = self.checks.${system}.pre-commit-check.enabledPackages;
+      };
+
+      formatter.${system} = pkgs.alejandra;
```

Comment:
@fricklerhandwerk please explain how, with this change, `nix fmt` will format YAML files. The way I understand this, only `alejandra` will be invoked, which will only format `*.nix` files.

Without this PR, using treefmt-nix, we could enable a YAML formatter in `treefmt.nix` and from that point on, `nix fmt` would format YAML files. The treefmt-nix formatter is currently registered (indirectly) in line 170, which is deleted in this PR as of a11678a.

Reply:
I see. @wegank can we hook treefmt into pre-commit, or any other way of having the best of both worlds?

Reply:
I've made `formatter.${system}` a shell script invoking `pre-commit` instead.